ccc-js/src/main/javascript/package-res/lib/protovis.js

```diff
@@ -4635,9 +4644,28 @@ DateComponent.prototype.ticks = function(min, max, mult, options) {
   return ticks;
 };
 
+DateComponent.prototype.setTickStart = function(tick, mult) {
+  if(this.tickStart) {
+    var diffPeriod = getDiff(this.tickStart, tick);
+    var multPeriod = this.mult * mult;
+    
+    while(diffPeriod > multPeriod) {
+      this.increment(this.tickStart, mult);
+      diffPeriod = getDiff(this.tickStart, tick);
+    }
+  } else {
+    this.tickStart = tick;
```

Comment:
As we'd talked about it, `tickStart` stored in the singleton `DateComponent` instance like you're proposing,  would affect every chart running on the same HTML page.
If you'd put multiple charts rendering multiple times during the same period, they'd run over each other's start ticks. Also, one chart rendered after another would incorrectly use the other's last start tick. That's why this "alignment tick" must be provided by each chart, as an option of the ticks method.